Code/GraphMol/MolAlign/Wrap/rdMolAlign.cpp

```diff
@@ -115,7 +125,45 @@ PyObject *getMolAlignTransform(const ROMol &prbMol, const ROMol &refMol,
     delete wtsVec;
   }
 
-  return generateRmsdTransPyTuple(rmsd, trans);
+  return generateRmsdTransMatchPyTuple(rmsd, trans);
+}
+
+PyObject *getBestMolAlignTransform(const ROMol &prbMol, const ROMol &refMol,
+                                   int prbCid = -1, int refCid = -1,
+                                   python::object map = python::list(),
+                                   int maxMatches = 1000000,
+                                   bool symmetrizeTerminalGroups = true,
+                                   python::object weights = python::list(),
+                                   bool reflect = false,
+                                   unsigned int maxIters = 50) {
+  std::vector<MatchVectType> aMapVec;
+  unsigned int nAtms = 0;
+  if (map != python::object()) {
+    aMapVec = translateAtomMapSeq(map);
+    if (!aMapVec.empty()) {
+      nAtms = aMapVec.front().size();
+    }
+  }
+  RDNumeric::DoubleVector *wtsVec = translateDoubleSeq(weights);
```

Comment:
How about making this a `unique_ptr` so that you don't have to worry about deallocation?

Reply:
I am all in favor of that, but the `delete` was there because I copy/pasted that code from another function in the same file, which is peppered with `delete` statements. So I got replaced all of them with `unique_ptr`s.

Reply:
thanks for that!